ewatercycle/config/_validators.py

```diff
@@ -0,0 +1,141 @@
+        # Allow any ordered sequence type -- generators, np.ndarray, pd.Series
+        # -- but not sets, whose iteration order is non-deterministic.
+        elif isinstance(inp,
+                        Iterable) and not isinstance(inp, (set, frozenset)):
+            # The condition on this list comprehension will preserve the
+            # behavior of filtering out any empty strings (behavior was
+            # from the original validate_stringlist()), while allowing
+            # any non-string/text scalar values such as numbers and arrays.
+            inp = [
+                scalar_validator(val) for val in inp
+                if not isinstance(val, str) or val
+            ]
+        else:
+            raise ValidationError(
+                f"Expected str or other non-set iterable, but got {inp}")
+        if n_items is not None and len(inp) != n_items:
+            raise ValidationError(f"Expected {n_items} values, "
+                                  f"but there are {len(inp)} values in {inp}")
+        return inp
+
+    try:
+        func.__name__ = "{}list".format(scalar_validator.__name__)
+    except AttributeError:  # class instance.
+        func.__name__ = "{}List".format(type(scalar_validator).__name__)
+    func.__qualname__ = func.__qualname__.rsplit(".",
+                                                 1)[0] + "." + func.__name__
+    if docstring is not None:
+        docstring = scalar_validator.__doc__
+    func.__doc__ = docstring
+    return func
+
+
+def validate_path(value, allow_none=False):
+    """Return a `Path` object."""
+    if (value is None) and allow_none:
+        return value
+    try:
+        path = Path(value).expanduser().absolute()
+    except TypeError as err:
+        raise ValidationError(f"Expected a path, but got {value}") from err
+    else:
+        return path
+
+
+validate_string = _make_type_validator(str)
+validate_string_or_none = _make_type_validator(str, allow_none=True)
+validate_stringlist = _listify_validator(validate_string,
+                                         docstring='Return a list of strings.')
+validate_int = _make_type_validator(int)
+validate_int_or_none = _make_type_validator(int, allow_none=True)
+validate_float = _make_type_validator(float)
+validate_floatlist = _listify_validator(validate_float,
+                                        docstring='Return a list of floats.')
+
+validate_path_or_none = _make_type_validator(validate_path, allow_none=True)
+
+validate_pathlist = _listify_validator(validate_path,
+                                       docstring='Return a list of paths.')
+
+_validators = {
+    'esmvaltool_config': validate_path_or_none,
+    'grdc_location': validate_path_or_none,
+    'container_engine': validate_string_or_none,
+    'singularity_dir': validate_path_or_none,
+    'output_dir': validate_path_or_none,
+    'ewatercycle_config': validate_path_or_none,
+    # wflow specific
+    'wflow.singularity_images': validate_string,
```

Comment:
I think I will leave it to @Peter9192 to update these values as necessary. I copied them from his PR to see if this could be a nice way to retrieve nested variables via `subset = CFG.get_subset('wflow')`.